### PR TITLE
Refinery migration support

### PR DESCRIPTION
**Pull request by wseaton**
Following the example in [klickhouse](https://github.com/Protryon/klickhouse/blob/master/klickhouse/src/migrate.rs) I've implemented support for [refinery](https://github.com/rust-db/refinery) embedded migrations.

I did this mainly because I wanted a rust core to build a migration tool to move away from schemachange.

Let me know if you'd prefer this move to another crate.

### CODE REVIEW

File: snowflake-api/src/migration.rs [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+
+#[async_trait]
+impl AsyncTransaction for SnowflakeApi {
+    type Error = SnowflakeApiError;
+
+    async fn execute(&mut self, queries: &[&str]) -> Result<usize, Self::Error> {
+        self.exec("BEGIN TRANSACTION").await?;
+
+        for query in queries {
+            let parsed_query = sqlparser::parser::Parser::parse_sql(&DIALECT, query)
```
Comment (andrusha) [inline snowflake-api/src/migration.rs]: from what I can see sqlparser dialect doesn't allow for put/get queries or any session control queries (alter session), it seem to follow public api, which is of limited use for this particular library

Comment (wseaton) [reply snowflake-api/src/migration.rs]: Ack yeah, you are totally right. Limited snowflake extended dialect coverage means this is worse than a more naiive impl of statement split.

**Comment by wseaton** [comment]
@andrusha still need to fix the merge conflict, but I ripped out the SQL parser and added some code as a workaround to fix an issue with assuming other roles during a migration potentially blocking your ability to write to the migration event table.

**Comment by dmzmk** [comment]
Hey @wseaton, thanks for contribution. IMO separate crate would be better in this case.

**Pull request closed without merging**